pkg/querier/plan/plan.go

```diff
@@ -8,6 +8,8 @@ import (
 
 type QueryPlan struct {
 	AST syntax.Expr
+
+	lineFilterExprs *[]syntax.LineFilterExpr
```

Comment:
QueryPlan should not contain individual parts of the AST separately.

Reply:
related: https://github.com/grafana/loki/pull/12037#discussion_r1499421814